test/utils/node-test.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * A port of tap's `t.type` that can be used with `node:assert`
+ * https://github.com/tapjs/tapjs/blob/511019b2ac0fa014370154c3a341a0e632f50b19/src/asserts/src/index.ts#L199
+ */
+function ttype (plan, obj, klass) {
```

Comment:
Let's not inject the `plan` object and rather make `ttype` return a `boolean` value so we are not coupling this `utils` to specific tools; otherwise will be harder to decuple or maintain. wdyt?

Reply:
Sounds good!

Reply:
Done [a874a48](https://github.com/nodejs/undici/pull/2564/commits/a874a48881c518f3d59b51a626603a445485a0e1)